x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/activity_log_date_range_picker/index.tsx

```diff
@@ -6,118 +6,142 @@
  */
 
 import { useDispatch } from 'react-redux';
-import React, { memo, useCallback } from 'react';
+import React, { memo, useCallback, useState } from 'react';
 import styled from 'styled-components';
-import moment from 'moment';
-import { EuiFlexGroup, EuiFlexItem, EuiDatePicker, EuiDatePickerRange } from '@elastic/eui';
+import dateMath from '@elastic/datemath';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSuperDatePicker,
+  EuiSuperDatePickerRecentRange,
+} from '@elastic/eui';
 
-import * as i18 from '../../../translations';
 import { useEndpointSelector } from '../../../hooks';
 import { getActivityLogDataPaging } from '../../../../store/selectors';
+import { DEFAULT_TIMEPICKER_QUICK_RANGES } from '../../../../../../../../common/constants';
+import { useUiSetting$ } from '../../../../../../../../public/common/lib/kibana';
+
+interface Range {
+  from: string;
+  to: string;
+  display: string;
+}
 
 const DatePickerWrapper = styled.div`
   width: ${(props) => props.theme.eui.fractions.single.percentage};
-  background: white;
+  max-width: 250px;
 `;
 const StickyFlexItem = styled(EuiFlexItem)`
-  max-width: 350px;
+  background: ${(props) => `${props.theme.eui.euiColorGhost}`};
   position: sticky;
-  top: ${(props) => props.theme.eui.euiSizeM};
+  top: 0;
   z-index: 1;
-  padding: ${(props) => `0 ${props.theme.eui.paddingSizes.m}`};
+  padding: ${(props) => `${props.theme.eui.paddingSizes.m}`};
 `;
 
 export const DateRangePicker = memo(() => {
   const dispatch = useDispatch();
-  const { page, pageSize, startDate, endDate, isInvalidDateRange } = useEndpointSelector(
-    getActivityLogDataPaging
-  );
+  const {
+    page,
+    pageSize,
+    startDate,
+    endDate,
+    autoRefreshOptions,
+    recentlyUsedDateRanges,
+  } = useEndpointSelector(getActivityLogDataPaging);
 
-  const onClear = useCallback(
-    ({ clearStart = false, clearEnd = false }: { clearStart?: boolean; clearEnd?: boolean }) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatchActionUpdateActivityLogPaging = useCallback(
+    async ({ start, end }) => {
+      setIsLoading(true);
       dispatch({
         type: 'endpointDetailsActivityLogUpdatePaging',
         payload: {
           disabled: false,
           page,
           pageSize,
-          startDate: clearStart ? undefined : startDate,
-          endDate: clearEnd ? undefined : endDate,
+          startDate: start ? dateMath.parse(start)?.toISOString() : undefined,
+          endDate: end ? dateMath.parse(end)?.toISOString() : undefined,
         },
       });
+      setIsLoading(false);
```

Comment:
I'm still not seeing the benefit of `setIsLoading()`. All of this here is happening in `sync` mode, so essentially, the UI does not even get rendered with `isLoading: true`, because we set it and unset it within this function.
Can you tie the `isLoading` to the actual API request/response?

Example:
You could have a selector like this:

```typescript
isDataLoading(dataLoadState) {
    return !isLoadedResourceState(dataLoadState)
}
```

And then the UI can just use that for "isLoading" instead of defining local state via `useState()`.

Reply:
Oh yes! 👍

Reply:
Okay, I realize `setIsLoading ` was needed for the update button's UI. Since we don't show the update button here this loading state is obsolete. I shall go ahead and remove it entirely.

Reply:
done d9662c5bb2857edcfa78a95d178e04635d599df5